solr/core/src/test/org/apache/solr/util/TestCircuitBreakers.java

```diff
@@ -66,26 +76,39 @@ protected static void indexDocs() {
   public void tearDown() throws Exception {
     super.tearDown();
     dummyMemBreaker.close();
-    dummyCBManager.close();
   }
 
   @After
   public void after() {
     removeAllExistingCircuitBreakers();
   }
 
-  public void testCBAlwaysTrips() {
+  public void testCBAlwaysTripsWithCorrectCode() {
     removeAllExistingCircuitBreakers();
 
     CircuitBreaker circuitBreaker = new MockCircuitBreaker(true);
 
     h.getCore().getCircuitBreakerRegistry().register(circuitBreaker);
 
-    expectThrows(
-        SolrException.class,
-        () -> {
-          h.query(req("name:\"john smith\""));
-        });
+    List.of(
+            SolrException.ErrorCode.TOO_MANY_REQUESTS.code,
+            SolrException.ErrorCode.SERVICE_UNAVAILABLE.code,
+            SolrException.ErrorCode.BAD_REQUEST.code)
+        .forEach(
+            code -> {
+              if (code != SolrException.ErrorCode.TOO_MANY_REQUESTS.code) {
+                System.setProperty(
+                    CircuitBreaker.SYSPROP_SOLR_CIRCUITBREAKER_ERRORCODE, String.valueOf(code));
+              }
+              SolrException ex =
+                  expectThrows(
+                      SolrException.class,
+                      () -> {
+                        h.query(req("name:\"john smith\""));
+                      });
+              assertEquals(code == -1 ? 429 : code, ex.code());
```

Comment:
question: when could the code be `-1` based on the list above?
```suggestion
              assertEquals(code == -1 ? SolrException.ErrorCode.TOO_MANY_REQUESTS.code : code, ex.code());
```

Reply:
Sorry, confusion here. I now went back to using -1 as "default", as `null` is not allowed in the list.

Reply:
```suggestion
              assertEquals(code == -1 ? SolrException.ErrorCode.TOO_MANY_REQUESTS.code : code, ex.code());
```